### PR TITLE
State transition form

### PR DESCRIPTION
**Pull request by AnshulShirude**
## Description/Problem

Closes [story](copy_trello_card_link_here)

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

## Solution

What did you change?
Added in the patch route for the state transition form.

## Dependencies

[ ] This PR adds new dependencies

## Testing
Did some of the manual testing and it works.

Please describe how you tested this PR (manually and/or automatically)
Provide instructions so we can reproduce, and include screenshots of UI changes.

- Manual Tests?
  - Add details and expected results here
- Unit/Integration/E2E Tests?
  - Add details and expected results here

### CODE REVIEW

File: apps/server/src/form-instances/form-instances.service.ts [hunk 1]
```diff
@@ -99,4 +99,31 @@ export class FormInstancesService {
...
     });
   }
+
+  async signFormInstance(formInstanceId: string, signatureId: string) {
+    const formInstance = await this.prisma.formInstance.findUnique({
+        where: { id: formInstanceId },
+        include: { signatures: { include: { signerPosition: true, userSignedBy: true } } },
+    });
+
+    if (!formInstance) {
```
Comment (elvincheng3) [inline apps/server/src/form-instances/form-instances.service.ts:109]: let's use the defined form instance error message, and throw a NotFoundException instead here

File: apps/server/src/form-instances/form-instances.service.ts [hunk 2]
```diff
@@ -99,4 +99,31 @@ export class FormInstancesService {
...
+        throw new Error('Form instance not found');
+    }
+
+    const signature = formInstance.signatures.find(sig => sig.id === signatureId);
+
+    if (!signature) {
+        throw new Error('Signature not found');
```
Comment (elvincheng3) [inline apps/server/src/form-instances/form-instances.service.ts:116]: same here, using the defined signature error message and a NotFoundException

File: apps/server/src/form-instances/form-instances.service.ts [hunk 3]
```diff
@@ -99,4 +99,31 @@ export class FormInstancesService {
...
+    const formInstance = await this.prisma.formInstance.findUnique({
+        where: { id: formInstanceId },
+        include: { signatures: { include: { signerPosition: true, userSignedBy: true } } },
+    });
+
+    if (!formInstance) {
+        throw new Error('Form instance not found');
+    }
+
+    const signature = formInstance.signatures.find(sig => sig.id === signatureId);
```
Comment (elvincheng3) [inline apps/server/src/form-instances/form-instances.service.ts:113]: we should verify that the signature is the next one to be signed (all previous signatures should be signed), or else throw an error

File: apps/server/src/form-instances/form-instances.service.ts [hunk 4]
```diff
@@ -99,4 +105,45 @@ export class FormInstancesService {
...
+      }
+    }
+
+    const updatedSignature = await this.prisma.signature.update({
+      where: { id: signatureId },
+      data: { signed: true },
+    });
+
+    formInstance.signatures[signatureIndex] = {
+      ...formInstance.signatures[signatureIndex],
```
Comment (elvincheng3) [inline apps/server/src/form-instances/form-instances.service.ts:143]: is this aggregating the existing signature with the new updates?

Comment (AnshulShirude) [reply apps/server/src/form-instances/form-instances.service.ts:143]: Yes, exactly. It's meant to update the signatures at the signatureIndex accordingly.

**Review by elvincheng3 — APPROVED** [review]

**Pull request merged**